examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+  U64,
+  ViewRequestData,
+} from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<void> => {
+  const payload: ViewRequestData = {
+    function: `0x${swap.toStringWithoutPrefix()}::router::optimal_liquidity_amounts`,
+    arguments: [
+      `0x${Hex.fromHexInput(token1Addr).toStringWithoutPrefix()}`,
+      `0x${Hex.fromHexInput(token2Addr).toStringWithoutPrefix()}`,
+      false,
+      "200000",
+      "300000",
+      "200",
+      "300",
+    ],
+  };
+  const result: any = await aptos.view({ payload });
+  console.log("Optimal LP amount: ", result);
+};
+
+const addLiquidity = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::add_liquidity_entry`,
+      arguments: [
+        AccountAddress.fromHexInput(token1Addr),
+        AccountAddress.fromHexInput(token2Addr),
+        new Bool(false),
+        new U64(200000),
+        new U64(300000),
+        new U64(200),
+        new U64(300),
+      ],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Add liquidity succeed. - ", response.hash);
+  return response.hash;
+};
+
+const swap = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  fromToken: HexInput,
+  toToken: HexInput,
+  amountIn: number,
+  amountOutMin: number,
+  recipient: AccountAddress,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::swap_entry`,
+      arguments: [
+        new U64(amountIn),
+        new U64(amountOutMin),
+        AccountAddress.fromHexInput(fromToken),
+        AccountAddress.fromHexInput(toToken),
+        new Bool(false),
+        recipient,
+      ],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Swap succeed. - ", response.hash);
+  return response.hash;
+};
+
+const getAssetType = async (aptos: Aptos, owner: Account): Promise<any> => {
+  const data = await aptos.getFungibleAssetMetadata({
+    options: {
+      where: {
+        creator_address: { _eq: owner.accountAddress.toString() },
+      },
+    },
+  });
+
+  if (data.length !== 2) throw new Error("Expected two Fungible Asset.");
```

Comment:
```suggestion
  if (data.length !== 2) throw new Error("Expected two Fungible Assets.");
```

Did you also want to verify the fungible assets are different?

I don't think a liquidity pool with the same exact assets make any sense?

Reply:
Yep correct. Hence I did created 2 different module (Cat and Dog) to separate out the two different fungible asset. 

Good call here, will check and make sure these two assets are differents.